File: app/src/main/java/com/otraempresa/android/example/AbstractRequestTransferTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

import com.kushkipagos.android.Card;
import com.kushkipagos.android.Kushki;
import com.kushkipagos.android.KushkiEnvironment;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;
import com.kushkipagos.android.Transfer;

abstract class AbstractRequestTransferTokenAsyncTask extends AsyncTask<Transfer, Void, Transaction> {

    protected final Kushki kushki;
    private final Context context;

    AbstractRequestTransferTokenAsyncTask(Context context) {
        this.context = context;
        kushki = new Kushki("10000002036955013614148494909956", "USD", KushkiEnvironment.TESTING);
    }

    @Override
    protected Transaction doInBackground(Transfer... args) {
        try {
            return requestTransferToken(args[0]);
        } catch (KushkiException kushkiException) {
            throw new RuntimeException(kushkiException);
        }
    }

    @Override
    protected void onPostExecute(Transaction transaction) {
        if (transaction.isSuccessful()) {
            showToast(transaction.getToken());
        } else {
            showToast("ERROR: " + transaction.getCode() + " " + transaction.getMessage());
        }
    }

    protected abstract Transaction requestTransferToken(Transfer card) throws KushkiException;

    private void showToast(String text) {
        Toast toast = Toast.makeText(context, text, Toast.LENGTH_SHORT);
        toast.show();
    }
}

File: app/src/main/java/com/kushkipagos/kushkidemo/contracts/AbstractRequestTransferTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.contracts

import android.content.Context
import android.widget.Toast
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushki.Kushki
import com.kushkipagos.kushki.KushkiEnvironment
import com.kushkipagos.kushkidemo.helpers.CoroutineTask
import com.kushkipagos.models.Transaction
import com.kushkipagos.models.Transfer

abstract class AbstractRequestTransferTokenAsyncTask: CoroutineTask<Transfer, Void, Transaction> {

    protected var _kushki: Kushki
    private var _context: Context

    constructor(context: Context){
        _context = context
        _kushki = Kushki("c308a8af32114b8c97f8ba191149df9b", "USD", KushkiEnvironment.QA)
    }

    @Throws(KushkiException::class)
    protected abstract fun requestTransferToken(token: Transfer): Transaction

    override fun doInBackground(vararg args: Transfer?): Transaction {
         try {
             return requestTransferToken(args[0]!!)
        } catch (kushkiException: KushkiException) {
            throw RuntimeException(kushkiException)
        }
    }

    override fun onPostExecute(param: Transaction?) {
        val transaction = param!!
        if (transaction.isSuccessful) {
            showToast(transaction.token)
        } else {
            showToast("ERROR: " + transaction.code + " " + transaction.message)
        }
    }

    fun showToast(text: String) {
        val toast = Toast.makeText(_context, text, Toast.LENGTH_SHORT)
        toast.show()
    }
}